panel/widgets/remote_file_selector.py

```diff
@@ -0,0 +1,379 @@
+
+    :Example:
+
+    >>> remote_file_provider = S3RemoteFileProvider(...)
+    >>> remote_file_selector = RemoteFileSelector(provider=remote_file_provider)
+
+    """
+
+    directory = param.String(default="/", doc="""
+        The directory to explore.""")
+
+    size = param.Integer(default=10, doc="""
+        The number of options shown at once (note this is the only
+        way to control the height of this widget)""")
+
+    value = param.List(default=[], doc="""
+        List of selected files.""")
+
+    _composite_type: ClassVar[Type[ListPanel]] = Column
+
+    _provider : RemoteFileProvider = None
+
+    _dir_prefix : str = '📁 '
+
+    # Used to navigate in the history of paths visited
+    _cache_cwd : List[str] = []
+    _history_position : int = -1
+
+    # associates a path (str) to a list of files (list of str)
+    _cache_filelist : Dict[str, List[str]] = {}
+
+    def __init__(self, provider: RemoteFileProvider, **params):
+
+        self._provider = provider
+
+        if params.get('width') and params.get('height') and 'sizing_mode' not in params:
+            params['sizing_mode'] = None
+
+        super().__init__(**params)
+
+        # Set up layout
+        layout = {p: getattr(self, p) for p in Layoutable.param
+                  if p not in ('name', 'height', 'margin') and getattr(self, p) is not None}
+        sel_layout = dict(layout, sizing_mode='stretch_width', height=300, margin=0)
+        self._selector = CrossSelector(
+            size=self.size, **sel_layout,
+        )
+
+        self._back = Button(name='◀', width=40, height=40, margin=(5, 10, 0, 0),
+                            align='center')
+        self._forward = Button(name='▶', width=40, height=40, margin=(5, 10, 0, 0),
+                                align='center')
+        self._up = Button(name='⬆', width=40, height=40, margin=(5, 10, 0, 0),
+                          align='center')
+        self._directory = TextInput(value=self.directory, margin=(5, 10, 0, 0), width_policy='max',
+                                    disabled=True)
+        self._down = Button(name='⬇', disabled=True, width=40, height=40, margin=(5, 5, 0, 0), align='center')
+        self._reload = Button(name='↻', width=40, height=40, margin=(5, 0, 0, 10), align='center')
+        self._nav_bar = Row(
+            self._back, self._forward, self._up, self._directory, self._down, self._reload,
+            **dict(layout, width=None, margin=0, width_policy='max')
+        )
+
+        self._composite[:] = [self._nav_bar, Divider(margin=0), self._selector]
+        style = 'h4 { margin-block-start: 0; margin-block-end: 0;}'
+        self._selector._selected.insert(0, Markdown('#### Selected files', margin=0, stylesheets=[style]))
+        self._selector._unselected.insert(0, Markdown('#### File Browser', margin=0, stylesheets=[style]))
+        self.link(self._selector, size='size')
+
+
+        # Set up state
+        self._stack = []
+        self._cwd = PurePosixPath("/")
```

Comment:
Is it always the case it is a PosixPath? 

There could be a RemoteFileProvider that connects to a Windows Server.